### PR TITLE
Fixed runtime error of extensions that depend on SlicerOpenIGTLink

### PR DESCRIPTION
**Pull request by lassoan**
In the build tree SlicerIGT modules that depended on SlicerOpenIGTLink extension failed to load because SlicerOpenIGTLinkConfig.cmake did not contain all necessary information.
Fixed it by removing custom SlicerOpenIGTLinkConfig.cmake generator code and just rely on include(${Slicer_EXTENSION_GENERATE_CONFIG}).

Also removed VP9 static libraries and headers files from the package. Static libraries are linked into executables and header files are not needed for runtime.

### CODE REVIEW

**Comment by leochan2009** [comment]
Hi Andras,

Thanks for your suggestion for the VP9 libraries.
Regarding the config file.
I tried use the "include(${Slicer_EXTENSION_GENERATE_CONFIG})" for config file. 
The config file generated by slicer would not link the dependent OpenIGTLink and OpenIGTLinkIO library correctly. 
So i looked into the example of [SlicerOpenCV](https://github.com/SBU-BMI/SlicerOpenCV.git), they have the same issue for linking dependent libraries. Their solution is to manual create the config file
(manual configuration)[https://github.com/SBU-BMI/SlicerOpenCV/blob/master/CMakeLists.txt#L66].
I will test your code and build slicerIGT against SlicerOpenIGTlink and check if the dependent libraries will be linked or not. 

Best,
Longquan

**Comment by leochan2009** [comment]
Hi Andras,

i tried use your branch, and build SlicerOpenIGTLink successfully.
However, the build of SlicerIGT failed..
The reason is as i mentioned, the OpenIGTLinkIO and OpenIGTlink libraries are not included in the SlicerOpenIGTLinkConfig.cmake file
See the error message i got by compiling SlicerIGT

**Comment by lassoan** [comment]
Thanks for the testing. I don't see this error but this is uncovered a problem: OpenIGTLink and OpenIGTLinkIO header files must not be listed in public header files, only in cxx or private headers.

For example, these must be moved to the cxx file:

    // OpenIGTLinkIO includes
    #include "igtlioConnector.h"
    #include "igtlioDeviceFactory.h"

I'll take care of these and update this pull request.

**Comment by leochan2009** [comment]
Hi Andras,

I think the issue is with the config file.
I was building SlicerIGT by setting the SlicerOpenIGTLink_DIR to the locally built SlicerOpenIGTLink.
Using slicer-generated SlicerOpenIGTLinkConfig.CMake file, the path to the Openigtlink and OpenIGTLinkIO libraries are not pointed out.
i would really appreciate that if you find a solution.

Best,
Longquan

**Comment by lassoan** [comment]
OK, I'll solve that, too. There is a variable that we can use to inject additional content into the generated config file.

File: CMake/SlicerBlockInstallVP9.cmake [hunk 1]
```diff
@@ -1,4 +1,4 @@
-if(Slicer_USE_VP9)
+if(Slicer_USE_VP9 AND NOT CMAKE_SYSTEM_NAME STREQUAL "Windows")
```
Comment (leochan2009) [inline CMake/SlicerBlockInstallVP9.cmake]: i am not sure why we should not install VP9 for windows system?

**Comment by lassoan** [comment]
.lib files are linked into DLLs, they are not needed runtime. Header files not needed runtime either. If we included them in the package then it would increase the package size very significantly.

**Comment by leochan2009** [comment]
Great! the slicerigt build succeeds too!

**Pull request merged**